drivers/dac/ad5791/ad5791.c

```diff
@@ -344,3 +344,68 @@ int32_t ad5791_setup(struct ad5791_dev *dev,
 
 	return status;
 }
+
+/***************************************************************************//**
+ * @brief SPI write to device using a mask.
+ *
+ * @param dev              - The device structure.
+ * @param register_address - Address of the register.
+ *                          Example:
+ *                          AD5791_REG_DAC          - DAC register
+ *                          AD5791_REG_CTRL         - Control register
+ *                          AD5791_REG_CLR_CODE     - Clearcode register
+ *                          AD5791_CMD_WR_SOFT_CTRL - Software control register
+ * @param mask	- The mask.
+ * @param value - The register data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ad5791_spi_write_mask(struct ad5791_dev *dev,
+			  uint8_t register_address,
+			  uint32_t mask,
+			  uint32_t value)
+{
+	int status;
+	uint32_t reg_data;
+
+	status = ad5791_get_register_value(dev, register_address, &reg_data);
+	if (status)
+		return status;
+
+	reg_data &= ~mask;
+	reg_data |= value;
+
+	return ad5791_set_register_value(dev, register_address, reg_data);
+}
+
+/***************************************************************************//**
+ * @brief SPI read from device using a mask.
+ *
+ * @param dev              - The device structure.
+ * @param register_address - Address of the register.
+ *                          Example:
+ *                          AD5791_REG_DAC          - DAC register
+ *                          AD5791_REG_CTRL         - Control register
+ *                          AD5791_REG_CLR_CODE     - Clearcode register
+ *                          AD5791_CMD_WR_SOFT_CTRL - Software control register
+ * @param mask	- The mask.
+ * @param value - Pointer to the register data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ad5791_spi_read_mask(struct ad5791_dev *dev,
```

Comment:
In my opinion, a spi_Read_mask function is useless and shouldn't be provided, as it simply does `result & mask` which is something a user could do.

This is contrary to spi_write_mask which is very useful.

Reply:
okay, will remove the spi_read_mask function